flink-core/src/test/java/org/apache/flink/api/java/typeutils/PojoTypeExtractionTest.java

```diff
@@ -877,4 +878,57 @@ public void testRecursivePojoWithTypeVariable() {
 		Assert.assertEquals(GenericTypeInfo.class, ((PojoTypeInfo) pti).getPojoFieldAt(0).getTypeInformation().getClass());
 	}
 
+	/**
+	 * POJO with a bounded generic parameter.
+	 *
+	 * @param <T> generic parameter bounded by one type
+	 */
+	public static class BoundedPojo<T extends Tuple2<Integer, String>> {
+
+		public T someKey;
+
+		public BoundedPojo() {}
+
+		public BoundedPojo(T someKey) {
+			this.someKey = someKey;
+		}
+	}
+
+	@Test
+	public void testBoundedPojos() {
+		TypeInformation<?> ti = TypeExtractor.createTypeInfo(BoundedPojo.class);
+		Assert.assertTrue(ti instanceof PojoTypeInfo);
+		PojoTypeInfo<?> pti = (PojoTypeInfo<?>) ti;
+		Assert.assertEquals(BoundedPojo.class, pti.getTypeClass());
+		Assert.assertTrue(pti.getPojoFieldAt(0).getTypeInformation() instanceof GenericTypeInfo);
+		TypeInformation<?> fti = pti.getPojoFieldAt(0).getTypeInformation();
+		Assert.assertEquals(Tuple2.class, fti.getTypeClass());
```

Comment:
nit: what about adding an error message for clarity? I mean this particular assert is the feature we want to address so I think it deserves an error message. The other asserts are self explanatory I think